src/commands/farm.rs

```diff
@@ -205,46 +229,91 @@ async fn subscribe_to_plotting_progress(
-                        let _ = summary.update(None, Some(total_farmed)).await; // ignore the error, since we will abandon this mechanism
-                        if is_initial_progress_finished.load(Ordering::Relaxed) {
-                            print!("\rYou have farmed {total_farmed} block(s) in total!");
-                            // use
-                            // carriage return to overwrite the current value
-                            // instead of inserting
-                            // a new line
-                            std::io::stdout().flush().expect("Failed to flush stdout");
-                            // flush the
-                            // stdout to make sure values are printed
-                        }
-                    }
+            .map_err(|err| eyre!("couldn't get events from node, because: {err}"))?;
+
+        for event in events {
+            // subscription is active when plotting is started, only print out rewards after
+            // plotting finishes to not corrup the progress bars
+            if is_initial_progress_finished.load(Ordering::Relaxed) {
+                match event {
+                    Event::Rewards(
+                        RewardsEvent::VoteReward { voter: author, reward }
+                        | RewardsEvent::BlockReward { block_author: author, reward },
+                    ) if author == reward_address.into() =>
+                        summary_update_values.maybe_new_reward = Some(Rewards(reward)),
+                    Event::Subspace(SubspaceEvent::FarmerVote {
+                        reward_address: author, ..
+                    }) if author == reward_address.into() =>
+                        summary_update_values.is_new_vote = true,
+                    _ => (),
                 }
-            })
-            .await;
+            }
+        }
+
+        if let Some(pre_digest) = new_block.pre_digest {
+            if pre_digest.solution.reward_address == reward_address {
+                summary_update_values.is_new_block_farmed = true;
+            }
+        }
+        let _ = summary.update(summary_update_values).await; // ignore the
+                                                             // error, we will
+                                                             // abandon this
+                                                             // mechanism
+        let (farmed_block_count, vote_count, total_rewards) = (
+            summary
+                .get_farmed_block_count()
+                .await
+                .context("couldn't read farmed block count value, summary was corrupted")?,
+            summary
+                .get_vote_count()
+                .await
+                .context("couldn't read vote count value, summary was corrupted")?,
+            summary
+                .get_total_rewards()
+                .await
+                .context("couldn't read total rewards value, summary was corrupted")?,
```

Comment:
I don't think you need to parse the file 3 times here. Summary is the smart wrapper around file, right? I think it might be worth redoing it. Maybe even removing it, as you can scan events from all blocks which are available to get all those stats

Reply:
summary logic is to be abandoned, so that's why i'm ok with hacky solutions and don't wanna invest much time. 
+1 for removing it, let us discuss how to query blocks to deduce `total rewards` and so on... I can do that in a later PR

Reply:
I would better remove it right now if it will be abandoned. There is no need to make even more logic around it.